libutils/sequence.c

```diff
@@ -449,3 +450,87 @@ int SeqStringLength(Seq *seq)
 
     return total_length;
 }
+
+// TODO: These static helper functions could be (re)moved
+static bool HasNulByte(const char *str, size_t n)
+{
+    for (int i = 0; i<n; ++i)
+    {
+        if (str[i] == '\0')
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+static long GetLengthPrefix(const char *data)
+{
+    if (HasNulByte(data, 10))
+    {
+        return -1;
+    }
+    if (!isdigit(data[0]))
+    {
+        return -1;
+    }
+    size_t length;
+    int ret = sscanf(data, "%10zd", &length);
+    if (ret != 1)
+    {
+        // incorrect number of items scanned
+        return -1;
+    }
+    return (long)length;
+}
+
+char *SeqStringSerialize(Seq *seq)
```

Comment:
Do we also need a function to write into a file? The resulting string may potentially be quite big.

Reply:
Yes, definitely.